dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/PhoneConfirmation.cshtml.cs

```diff
@@ -54,7 +54,7 @@ public async Task<IActionResult> OnPost()
         var permittedUserTypes = authenticationState.GetPermittedUserTypes();
 
         var user = await _dbContext.Users
-            .Where(u => (u.MobileNumber ?? "") == PhoneHelper.FormatMobileNumber(MobileNumber!))
+            .Where(u => u.NormalizedMobileNumber! == parsedMobileNumber)
```

Comment:
Does this work? I was afraid of asserting non-null here

Reply:
It does work, indeed it was the only way I could make the nullable annotations happy (without `#pragma'-ing stuff).